extensions/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java

```diff
@@ -240,24 +243,30 @@ public void build(
         }
 
         Map<String, String> resteasyInitParameters = new HashMap<>();
+        ResteasyDeployment deployment = new ResteasyDeploymentImpl();
 
-        registerProviders(resteasyInitParameters, reflectiveClass, unremovableBeans, jaxrsProvidersToRegisterBuildItem);
+        registerProviders(deployment, resteasyInitParameters, reflectiveClass, unremovableBeans,
+                jaxrsProvidersToRegisterBuildItem);
 
         if (!resources.isEmpty()) {
+            deployment.getScannedResourceClasses().addAll(resources);
             resteasyInitParameters.put(ResteasyContextParameters.RESTEASY_SCANNED_RESOURCES, String.join(",", resources));
```

Comment:
do we really need to set both the init parameters and the deployment values?

Reply:
Yes.  The init parameters will eventually disappear.  That's work that is scheduled.